server/src/internalClusterTest/java/org/elasticsearch/cluster/ClusterHealthIT.java

```diff
@@ -311,4 +312,34 @@ public void testHealthOnMasterFailover() throws Exception {
             assertSame(responseFuture.get().getStatus(), ClusterHealthStatus.GREEN);
         }
     }
+
+    public void testWaitForEventsTimesOutIfMasterBusy() {
+        final AtomicBoolean keepSubmittingTasks = new AtomicBoolean(true);
+        final ClusterService clusterService = internalCluster().getInstance(ClusterService.class, internalCluster().getMasterName());
+        clusterService.submitStateUpdateTask("looping task", new ClusterStateUpdateTask(Priority.LOW) {
```

Comment:
We could maybe use a `PlainActionFuture<Void>` here and resolve it if `keepSubmittingTasks.get()` is `false` or on failure. Then we can just `.get()` on it at the end of the test and not leak the update task beyond the test time? (also I think we could get some strange test failures without that, where we set `false` for keep submitting, then leak the task, and get an `onFailure` when we shut down the cluster after the test?)

Reply:
Good point, yes. Also in the other test that submits a looping cluster state task - we're waiting for it to complete assuming everything else goes smoothly but otherwise we don't clean it up, so I adjusted that too. See 40945b7.